### PR TITLE
[K2] Enable Java tests for symbols

### PR DESCRIPTION
**Pull request by vmishenev**
It is based on #3202. Otherwise, we will have the flaky tests.
Is is tested by https://github.com/Kotlin/dokka/actions/runs/6486414842

### CODE REVIEW

File: plugins/base/src/test/kotlin/content/params/ContentForParamsTest.kt [hunk 1]
```diff
@@ -805,7 +801,7 @@ class ContentForParamsTest : BaseAbstractTest() {
             | /**
             | * a normal comment
             | *
-            | * @return empty string when [Hash Map](java.util.HashMap.containsKey) doesn't contain value.
+            | * @return empty string when [Hash Map][java.util.HashMap.containsKey] doesn't contain value.
```
Comment (IgnatBeresnev) [inline plugins/base/src/test/kotlin/content/params/ContentForParamsTest.kt:804]: Out of curiosity: did you change it because the k2 tests failed or was it an unrelated fix while passing by?

Comment (vmishenev) [reply plugins/base/src/test/kotlin/content/params/ContentForParamsTest.kt:804]: The former. `[Hash Map](java.util.HashMap.containsKey)` is an incorrect link.

Comment (vmishenev) [reply plugins/base/src/test/kotlin/content/params/ContentForParamsTest.kt:804]: See https://kotlinlang.org/docs/kotlin-doc.html#links-to-elements

Comment (IgnatBeresnev) [reply plugins/base/src/test/kotlin/content/params/ContentForParamsTest.kt:804]: yeah, I understand, but it doesn't answer the question :) Does K1 support such incorrect links?

Comment (vmishenev) [reply plugins/base/src/test/kotlin/content/params/ContentForParamsTest.kt:804]: > Does K1 support such incorrect links?

Yes, that is why the test was passed.
But IDE and K2 do not support them. 
image
